### ISSUE TITLE
idea

### ISSUE THREAD

**Issue by tj** [koajs/help#1]
the idea of this is to allow users to simply ask questions and find help quickly, since the koa eco-system is modular by design but also very spread out, and we all know how brutal it is to find things on npm. 

For example `$ koa how do I parse request bodies` would just filter the stop words and respond with some suggestions, or just `koa parse request` etc. I also have a wiki scraper that I built for clibs so we can scrape the Koa/Co wikis to respond with results as well as any canned responses. This will hopefully help guide people on their way without having to search around the site and wiki(s) too much.

Also could double as the manpage and api search, `koa ctx.host` etc

**Comment by hallas** [comment]
lets give it a shot

**Comment by tj** [comment]
:D for now maybe we just fetch/parse the [API docs](https://github.com/koajs/koa/tree/master/docs/api) with marked, to index each section, we'd have a lot of text to work off of there. The ecosystem stuff could come later

**Comment by hallas** [comment]
yep, keyword index the sections and we should be good to go, you wonna have search on the website? or just cli?

**Comment by tj** [comment]
nah I just fine executables way more handy than sites personally, I get annoyed when I have to go look at nodejs.org I wish the manpage had more stuff. Maybe some day we can add search to the site but it's just gh-pages for now

**Comment by hallas** [comment]
cool

**Comment by tj** [comment]
I guess the hard part will be getting people aware of this module haha, a site search would definitely better for that

**Comment by hallas** [comment]
we fix the executable and set it up so the site uses the executable to do the search, if we don't have it on the site, chances are people are gonna use the issues as an mailing group, which just sucks

**Comment by tj** [comment]
yea if we write it in a lib-ish way we can reuse it for site/cli/irc bot

**Comment by hallas** [comment]
your or I start :) ?

**Comment by tj** [comment]
I won't have time for a while, huge backlog :(

**Comment by hallas** [comment]
I'd like to start out then, you can always review, we'll see if it fits our needs, either way I consider it a good exercise

**Comment by hallas** [comment]
do you know how I can fork an empty project?

**Comment by tj** [comment]
no clue, I'll add a readme

**Comment by jonathanong** [comment]
is this going to just be an executable or a guide/discussion repo + executable?

**Comment by jonathanong** [comment]
we can have the executable search github issues. maybe tag the relevant ones to give them higher priority.

**Comment by tj** [comment]
I was just planning on it being the executable, or maybe a lib+exec since most people would probably prefer the site having this functionality I guess
